geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/coprocessor/KryoLazyDensityCoprocessor.scala

```diff
@@ -120,18 +128,17 @@ class KryoLazyDensityCoprocessor extends KryoLazyDensityService with Coprocessor
         throw cnfe
       case dse: DeserializationException =>
         throw dse
-    } finally CloseQuietly(scanner)
+    }
 
     done.run(response)
   }
 }
 
-object KryoLazyDensityCoprocessor extends KryoLazyDensityUtils {
+object KryoLazyDensityCoprocessor {
```

Comment:
i think the problem with the KryoLazyDensityUtils was that it needs to be used in instance classes, not objects. It has mutable state. So possibly just making the class instead of the object extend it would fix the issues you saw. I think the problem with the current changes is that the Z2/3 density iterators are not overriding the methods correctly anymore